lib/client-assets.php

```diff
@@ -432,47 +432,52 @@ function gutenberg_register_vendor_script( $handle, $src, $deps = array() ) {
 }
 
 /**
- * Extend wp-api Backbone client with methods to look up the REST API endpoints for all post types.
+ * Extend wp-api Backbone client with methods to look up Backbone Model and Collection by type
  *
- * This is temporary while waiting for #41111 in core.
+ * This is temporary while waiting for #42737 in core.
  *
- * @link https://core.trac.wordpress.org/ticket/41111
+ * @link https://core.trac.wordpress.org/ticket/42737
  */
 function gutenberg_extend_wp_api_backbone_client() {
-	// Post Types Mapping.
-	$post_type_rest_base_mapping               = array();
-	$post_type_rest_base_mapping['attachment'] = 'media';
-	$post_type_rest_base_mapping               = apply_filters( 'post_type_rest_base_mapping', $post_type_rest_base_mapping );
-
-	// Taxonomies Mapping.
-	$taxonomy_rest_base_mapping = array();
-	foreach ( get_taxonomies( array(), 'objects' ) as $taxonomy_object ) {
-		if ( ! empty( $taxonomy_object->rest_base ) ) {
-			$taxonomy_rest_base_mapping[ $taxonomy_object->name ] = $taxonomy_object->rest_base;
-		}
-	}
-	$taxonomy_rest_base_mapping = apply_filters( 'taxonomy_rest_base_mapping', $taxonomy_rest_base_mapping );
 
-	$script  = sprintf( 'wp.api.postTypeRestBaseMapping = %s;', wp_json_encode( $post_type_rest_base_mapping ) );
-	$script .= sprintf( 'wp.api.taxonomyRestBaseMapping = %s;', wp_json_encode( $taxonomy_rest_base_mapping ) );
+	// Type to Model Mapping, only need to map types that name differs from model.
+	// For example, no need to map post => post, or page => page.
+	$type_model_mapping               = array();
+	$type_model_mapping['attachment'] = 'media';
+	$type_model_mapping['post_tag']   = 'tag';
+	// You can use the filter to extend or map a custom type to another.
```

Comment:
Not sure if I am following, but maybe the above could be:

```php
$type_model_mapping = array(
  'attachment' => 'media',
  'post_tag'   => 'tag',
);
```

Same for `$type_collection_mapping`?

Reply:
> I don't really like the mapping of values by hand done here, so welcome any suggestions.

I think manual is the way to go, it has to be coded somewhere. Maybe there is something in the `WP_Post_Type` object that already contains this information, and we could use that to auto-magically map everything?